src/mlpack/core/cv/metrics/roc_auc_score_impl.hpp

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file core/cv/metrics/roc_auc_score_impl.hpp
+ * @author Sri Madhan M
+ *
+ * Implementation of the area under Receiver Operating Characteristic curve
+ * (ROC-AUC) score.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_METRICS_ROCAUCSCORE_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_ROCAUCSCORE_IMPL_HPP
+
+namespace mlpack {
+namespace cv {
+
+template<size_t PositiveClass>
+double ROCAUCScore<PositiveClass>::Evaluate(const arma::Row<size_t>& labels,
+                                            const arma::Row<double>& scores)
+{
+  util::CheckSameSizes(labels, scores, "ROCAUCScore::Evaluate()");
+
+  if (labels.n_cols == 0) {
+    throw std::invalid_argument(
+        "ROCAUCScore::Evaluate(): "
+        "number of points in input data cannot be zero");
+  }
+
+  // Compute labels with "1" for positive class and "0" for the other.
+  arma::Col<size_t> binaryLabels = arma::conv_to<arma::Col<size_t>>::from(
+      (labels == PositiveClass));
+
+  // Converting probability scores of PositiveClass, from row to column vector.
+  arma::Col<double> colScores = arma::conv_to<arma::Col<double>>::from(scores);
+
+  size_t numberOfTrueLabels  = arma::sum(binaryLabels);
+  size_t numberOfFalseLabels = binaryLabels.n_rows - numberOfTrueLabels;
+
+  // Check if only one class is given in labels.
+  if (numberOfTrueLabels == 0 || numberOfFalseLabels == 0) {
+    throw std::invalid_argument(
+        "ROCAUCScore::Evaluate(): "
+        "only one class is given in labels, ROCAUCScore is undefined");
+  }
+
+  // Sort labels and probabilities, using probability scores.
+  arma::ucolvec sortedScoreIndices = arma::stable_sort_index(
+      colScores, "descend");
```

Comment:
```suggestion
  arma::uvec sortedScoreIndices = arma::stable_sort_index(colScores, "descend");
```
This should be the same thing.